src/randomPicker.ts

```diff
@@ -41,6 +41,46 @@ export class RandomPicker<T> {
     return items;
   }
 
+  /**
+   * picks distinct elements from the weighted array.
+   * If the number of items required is more or equal to the length of weighted array, it will return all elements in array
+   * @param amount - positive number of items to be picked, if more or equal to the length of the array, returns all items
+   * @returns array of picked items
+   * @example
+   * const items = [
+   * { original: "Bronze", weight: 20 },
+   * { original: "Silver", weight: 10 },
+   * { original: "Gold", weight: 3 },
+   * { original: "Platinum", weight: 1 },
+   * ];
+   * const pickerAlwaysLast = new RandomPicker(items);
+   * const picked1 = pickerAlwaysLast.pickManyDistinct(1);
+   * console.log(picked1.length) // 1
+   * const picked3 = pickerAlwaysLast.pickManyDistinct(3);
+   * console.log(picked3.length) // 3
+   * const picked4 = pickerAlwaysLast.pickManyDistinct(4);
+   * console.log(picked4.length) // 4
+   * console.log(picked4) // ["Bronze", "Silver", "Gold", "Platinum"]
+   * const picked8 = pickerAlwaysLast.pickManyDistinct(8);
+   * console.log(picked8.length) // 4
+   * console.log(picked8) // ["Bronze", "Silver", "Gold", "Platinum"]
+   */
+  pickManyDistinct(amount: number): T[] {
+    if (amount < 0)
+      throw new Error(
+        "number of items to be picked should be a positive integer"
+      );
+    if (amount >= this.items.length) return this.items.map((i) => i.original);
```

Comment:
This means if there are 3 items and I want to pick 5, I will get 3 items instead.

What do you think about throwing an error if `amount >= this.items.length`?

If `amount === this.items.length` it's indeed a good idea to just return everything, I didn't think about it :D